lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -799,6 +823,79 @@ spec = do
         expectResponseCode HTTP.status403 rq
         expectErrorMessage (errMsg403WrongPool wrongPoolId) rq
 
+    it "STAKE_POOLS_JOIN/QUIT - Checking delegation expectations" $ \ctx -> do
+        (_, p1:p2:_) <- eventually $
+            unsafeRequest @[ApiStakePool] ctx Link.listStakePools Empty
+        w <- fixtureWallet ctx
+
+        request @ApiWallet ctx (Link.getWallet @'Shelley w) Default Empty >>= flip verify
+            [ expectFieldEqual #delegation notDelegatingNotJoining
+            ]
+
+        -- joining first pool
+        r1 <- joinStakePool ctx (p1 ^. #id) (w, fixturePassphrase)
+        expectResponseCode HTTP.status202 r1
+        eventually $ do
+            let ep = Link.listTransactions @'Shelley w
+            request @[ApiTransaction n] ctx ep Default Empty >>= flip verify
+                [ expectListItemFieldEqual 0 (#direction . #getApiT) Outgoing
+                , expectListItemFieldEqual 0 (#status . #getApiT) InLedger
+                ]
+        (currentEpochNo1, sp1) <- getSlotParams ctx
+        request @ApiWallet ctx (Link.getWallet @'Shelley w) Default Empty >>= flip verify
+            [ expectFieldEqual #delegation
+                (notDelegatingJoining (p1 ^. #id) currentEpochNo1 sp1)
+            ]
+        eventually $ do
+            request @ApiWallet ctx (Link.getWallet @'Shelley w) Default Empty >>= flip verify
+                [ expectFieldEqual #delegation (delegating (p1 ^. #id))
+                ]
+
+        -- rejoining second pool
+        r2 <- joinStakePool ctx (p2 ^. #id) (w, fixturePassphrase)
+        expectResponseCode HTTP.status202 r2
+        eventually $ do
+            let ep = Link.listTransactions @'Shelley w
+            request @[ApiTransaction n] ctx ep Default Empty >>= flip verify
+                [ expectListItemFieldEqual 0 (#direction . #getApiT) Outgoing
+                , expectListItemFieldEqual 0 (#status . #getApiT) InLedger
+                , expectListItemFieldEqual 1 (#direction . #getApiT) Outgoing
+                , expectListItemFieldEqual 1 (#status . #getApiT) InLedger
```

Comment:
Maybe clarify why there are two transactions here.

Reply:
After rejoining I want to discover **two** certificates (which were communicated to blockchain as txs).
Likewise, after quitting I am waiting for another one (so want to have **three**)

Reply:
Sorry, meant in a comment. And maybe you could say which his the most recent one? (The 0th one, right?)

Or… perhaps it would just be cluttering…